Fix 1 cent tax rounding issue

KRens opened the pull request:
For certain tax rates, eg: 21%, the pricing is wrong and illegal.
Eg: user paid 7€
If I fill in unit cost 5.78, then the invoice is 6,99€.
If I enter unit cost 5.79, then the invoice is 7,01€
No way to get to 7€.
This fixes it for me.

codedge commented:
Thanks for the contrib, @KRens!

I got a couple of questions and remarks:
- Next time, please see our [Contribution guide](https://github.com/invoiceninja/invoiceninja/blob/master/CONTRIBUTING.md) first.
  - Only PRs against `develop` are accepted.
- Why do you sometimes use 3 and sometimes 4 decimals?
- Why do you also adjust the qty to 3 decimals?
- The function you adjusted is called `roundToTwo`. It would have been better to create a function `roundToFour` and put in your code there.


As it is now, we cannot merge this in, I think. What do you think @hillelcoren ?

KRens commented:
- Sorry, first time I commit on something third party.
Will read contribution guide first next time.
- I started with rounding to 3, but then I still had issues with some invoices.
In the end I changed some to 4, just until everything started working.
Would you like me to test it again some time with 4 everywhere?
- Probably qty doesn't need adjusting to 3 decimals. I don't know your code properly, so I just changed everything.
- A function "roundToFour" already exists, but just replacing everywhere "roundToTwo" with "roundToFour" dit not work, as you applied some extra functions on the last one.
They are not identical. Tried it and it didn't work.
I could rename "roundToTwo" with something else, or make a new function, but I was afraid I would open pandora's box.
As far as my tests goes, changing it globally to rounding to 4 should work and should not cause any issues. Users who don't want to use it can simply ignore it and type only 2 digits.

codedge commented:
- Not a problem, but keep it mind. It's not only us that have contribution rules, also other projects ;-) In any case this PR cannot be merged as it is against `master` branch.
- In general I am not sure about this change. The rounding topic is very sensitive and requires thorough testing. Furthermore this whole topic play into the one explained here: https://github.com/invoiceninja/invoiceninja/issues/252

I am totally with you, that there should be some improvements in IN - I'd suggest one of the core dev looks into the topic.

hillelcoren commented:
Thanks for creating the pull request!

Sorry we can't merge it, hopefully other people find it useful.
